### DISCUSSION TITLE
NSFW filter toggle [FEATURE REQUEST]

### DISCUSSION THREAD

**Discussion by MontyCristo** [hlky/stable-diffusion-webui#218] [Feature suggestions]
Could you please add a toggle to turn on the NSFW filter? You have the best GUI I've seen and I would really love to use it if there was a config option to turn that on.

**Comment by hlky** [comment]
NSFW filter is on `diffusers` codebase. will have to check if it can be easily ported

**Comment by altryne** [comment]
@MontyCristo you want to ENABLE that toggle to get blurred results? 
hmmm maybe if this UI is going to get used in collab and save images to google drive that's not a bad idea 🤔

@hlky could this be helpful? 
https://github.com/CompVis/stable-diffusion/blob/main/scripts/txt2img.py#L87-L94

**Comment by MontyCristo** [comment]
@altryne Yes, that's right. I would really like to ENABLE the blurring or blacking filter. I'd love to be using this tech for some family friendly uses.

**Comment by SweetSophia** [comment]
> @altryne Yes, that's right. I would really like to ENABLE the blurring or blacking filter. I'd love to be using this tech for some family friendly uses.

You could use prompts to enable some kind of nsfw filter. You can exclude certain stuff from images.

**Reply by MontyCristo** [reply]
I didn't know that - is there a certain syntax used for excluding keywords?

**Reply by telemole** [reply]
> > @altryne Yes, that's right. I would really like to ENABLE the blurring or blacking filter. I'd love to be using this tech for some family friendly uses.
> 
> You could use prompts to enable some kind of nsfw filter. You can exclude certain stuff from images.

Would also love to know what to insert into prompts to remove NSFW results - I could hardcode to add those tags every time, pretty quick way to filter if there are keywords that work? using the [-word] format?

**Comment by MontyCristo** [comment]
Does anyone know the syntax/does it exist to exclude keywords?

**Comment by SweetSophia** [comment]
Not excluding per se, but you can use weights to minimize the risk of
nsfw/nudity stuff greatly.

MontyCristo ***@***.***> schrieb am Mi., 7. Sept. 2022, 22:44:

**Comment by grmrgecko** [comment]
Submitted pull request https://github.com/sd-webui/stable-diffusion-webui/pull/955 - in my testing, the NSFW content that randomly pops up are not filterable by weights, but I may be missing something. I can give examples if requested.
